awswrangler/athena/_write_iceberg.py

```diff
@@ -269,6 +274,10 @@ def to_iceberg(
         If none is provided, the AWS account ID is used by default
     schema_evolution: bool
         If True allows schema evolution for new columns or changes in column types.
+        Missing columns will throw an error unless ``schema_fill_missing`` is set to ``True``.
+    schema_fill_missing: bool
+        If True, fill missing columns with NULL values.
+        Only takes effect if ``schema_evolution`` is set to True.
```

Comment:
Do I understand correctly that if `schema_evolution=True` and this is `False`, then the table will be updated such that columns missing in the df are dropped from the table, and if `True`, they will be added and filled with `None`?

If so, this request, together with #2606 probably makes the case for `mode={append, overwrite}` kwarg, similar to the  `wr.s3.to_parquet` i.e. if the mode is `append` the data will be added to the table (and missing columns filled with `NULL`), and `overwrite` would overwrite the data and update the schema. Wyt?

Reply:
Just to make sure we're on the same page, your proposal is for the behavior to be the following, if a DataFrame came in that didn't match the current Iceberg schema:
1. `schema_evolution=False` and `mode="append"` - throw error saying schema doesn't match
2. `schema_evolution=False` and `mode="overwrite"` - just overwrite the entire table
3. `schema_evolution=True` and `mode="append"` - make the required changes to the schema
4. `schema_evolution=True` and `mode="overwrite"` - same behavior as 2.

The only thing I'm not sure of is what happens under scenario 1 if the only difference between the two schemas is that the DF is missing certain columns. Do we just fill them with None, or throw an error? I believe the former is the behavior for `wr.s3.to_parquet`.

Reply:
Hi,
Please allow me to comment.
I was trying to save pyspark DF to Iceberg table.
https://iceberg.apache.org/docs/latest/spark-writes/#writing-with-dataframes

`df.writeTo(path).option("mergeSchema","true").append()`

I was checking 2 scenarios:

- Saving DF with missing _column_a_  - result is that missing _column_a_ was filled with `null` value in the table
- Saving DF with the new additional _column_b_ - result is that table has new additional _column_b_, but the previous data set is filled with `null` for that column

I use `wr.s3.to_parquet` also and the behavior is the same as I described above.

I would expect that `to_iceberg` will return the same results with needed configuration setup.